### PR TITLE
move fflush to a thread

### LINKED ISSUE #15: Wurlitzer hangs process when pipe/buffer is full
Hi,

We're using Wurlitzer at Comet.ml and we found that when the C library prints a large string
the entire process hangs. We manage to create a test that reproduces this. Would it be possible to increase the buffer size or find a workaround?

```
def test_buffer_full():
    with pipes(stdout=PIPE, stderr=PIPE) as (stdout, stderr):
        long_string = "x" * 1000000 # create a very long string
        printf_err(long_string)

    # Test never reaches here as the process hangs.
    assert stderr.read() == long_string+"\n"
```

### PR DESCRIPTION
**Pull request by minrk**
fflush can block if buffers are full

include regression test for large output

closes #15

### CODE REVIEW

File: wurlitzer.py [hunk 1]
```diff
@@ -155,17 +155,25 @@ def forwarder():
...
             done = False
             draining = False
+            flush_interval = 0
             while pipes:
-                # flush libc's buffers before calling select
-                libc.fflush(c_stdout_p)
-                libc.fflush(c_stderr_p)
-                r, w, x = select.select(pipes, [], [], self.flush_interval)
-                if not r:
+                r, w, x = select.select(pipes, [], [], flush_interval)
```
Comment (gidim) [inline wurlitzer.py]: Why did you change self.flush_interval to flush_interval?

Comment (minrk) [reply wurlitzer.py]: Because now it may be self._flush_interval or 0 depending on the previous state.
